app/controllers/orders_controller.rb

```diff
@@ -74,8 +72,12 @@ def filter_params
     @filter_params ||= params[:filter]&.slice(:customer, :product, :shipper, :employee, :min_total_price, :max_total_price)
   end
 
+  def set_order
+    @order = Order.find(params[:id])
+  end
+
   def set_order_with_total_price
-    @order = Order.with_total_price.find(params[:id])
+    @order = Order.with_total_price.includes(:customer, :employee, :shipper, order_details: :product).find(params[:id])
```

Comment:
Wcześniej użyłaś eager_load() a tutaj includes(). Intencjonalnie?

Reply:
Opisałam to w https://github.com/klaudia-janiec/northwind-rails/wiki/Eager-loading
`eager_load` robi zawsze left outer joina
`includes` robi zazwyczaj jedno query na każdą relację
Potestowałam tam i wyszło mi że `eager_load` jest lepszy kiedy jest mniej relacji a `includes` wtedy kiedy więcej. Ale nie są to jakieś straszne różnice - parę ms. W aplikacji webowej parę milisekund nie ma znaczenia, no ale tak kazali :woman_shrugging: